### PR TITLE
#2732: Hunqz crawler and parser support

### PR DESCRIPTION
**Pull request by HanClinto**
Adds crawler and parser for Hunqz as per 2732

### CODE REVIEW

File: sites/hunqz_com/ad_parser.py [hunk 1]
```diff
@@ -0,0 +1,113 @@
...
+        self.page = html
+        try:
+            self.json = json.loads(html)
+        except JSONDecodeError:
+            self.json = {}
+        super().__init__(html)
+
+    def is_ad(self):
+        return "name" in self.json
+
```
[the remark this answers is not in the document]
Comment (HanClinto) [reply sites/hunqz_com/ad_parser.py:25]: Okay, I believe these have all been fixed in 6948736.  Thank you!

File: sites/hunqz_com/tests/ad_parser/20211025_0.html [hunk 2]
```diff
@@ -0,0 +1,125 @@
+{
```
[the remark this answers is not in the document]
Comment (HanClinto) [reply sites/hunqz_com/tests/ad_parser/20211025_0.html]: Completed in 6948736

File: sites/hunqz_com/ad_parser.py [hunk 3]
```diff
@@ -0,0 +1,113 @@
...
+        ad_text["is_plus"] = self._get("is_plus")
+        ad_text["last_login"] = self._get("last_login")
+        ad_text["location"] = self._get("location")
+        ad_text["personal"] = self._get("personal")
+        ad_text["service"] = self._get("service")
+        ad_text["sexual"] = self._get("sexual")
+        ad_text["telephone"] = self._get("telephone")
+        ad_text["creation_date"] = self._get("creation_date")
+
+        return json.dumps(ad_text)
```
[the remark this answers is not in the document]
Comment (HanClinto) [reply sites/hunqz_com/ad_parser.py]: Completed in 6948736.  Thank you!

File: sites/hunqz_com/ad_parser.py [hunk 4]
```diff
@@ -0,0 +1,113 @@
...
+        return self._get("personal","orientation").lower()
+
+    def _get(self, *keys):
+        val = self.json
+        for key in keys:
+            if key in val:
+                val = val[key]
+            else:
+                return ""
+        return val
```
[the remark this answers is not in the document]
Comment (HanClinto) [reply sites/hunqz_com/ad_parser.py]: Thank you!  Ultimately I elected not to use this, but I am glad to know that this is there.

File: sites/hunqz_com/ad_parser.py [hunk 5]
```diff
@@ -0,0 +1,113 @@
...
+    def is_ad(self):
+        return "name" in self.json
+
+    def primary_phone_number(self):
+        return self._get("telephone")
+
+    # An array of all phone numbers found in the ad using regex.
+    # Ex: [(347) 305-2188, 7864689700, 3473052188]
+    def phone_numbers(self):
+        return []
```
[the remark this answers is not in the document]
Comment (HanClinto) [reply sites/hunqz_com/ad_parser.py]: Added in commit 6948736.  Thank you!

File: sites/hunqz_com/ad_parser.py [hunk 6]
```diff
@@ -0,0 +1,113 @@
...
+
+    def name(self):
+        return self._get("name")
+
+    def primary_email(self):
+        return None
+
+    # An array of all emails found in the ad using regex.
+    def emails(self):
+        return []
```
[the remark this answers is not in the document]
Comment (HanClinto) [reply sites/hunqz_com/ad_parser.py]: Added in commit 6948736.  Thank you!

File: sites/hunqz_com/ad_parser.py [hunk 7]
```diff
@@ -0,0 +1,113 @@
...
+    def primary_phone_number(self):
+        return self._get("telephone")
+
+    # An array of all phone numbers found in the ad using regex.
+    # Ex: [(347) 305-2188, 7864689700, 3473052188]
+    def phone_numbers(self):
+        return []
+
+    def date_posted(self):
+        # Only dates available are last_login and creation_date.
```
[the remark this answers is not in the document]
Comment (peoplespete) [reply sites/hunqz_com/ad_parser.py:42]: I just switched to creation_date. I'm just thinking if account creation is the same as ad creation, then that feels like what date_posted is meant to capture.

**Pull request merged**